dvc/output.py

```diff
@@ -788,18 +793,14 @@ def unprotect(self):
             self.odb.unprotect(self.path_info)
 
     def get_dir_cache(self, **kwargs):
-
         if not self.is_dir_checksum:
             raise DvcException("cannot get dir cache for file checksum")
 
         try:
-            objects.check(self.odb, self.odb.get(self.hash_info))
+            obj = self.odb.get(self.hash_info)
```

Comment:
Let's move it out of `try`, just to avoid making the `except` confusing because of `obj`  being used in both.